geonode/harvesting/tasks.py

```diff
@@ -111,11 +111,8 @@ def _harvest_resource(
     harvestable_resource = models.HarvestableResource.objects.get(
         pk=harvestable_resource_id)
     worker = harvestable_resource.harvester.get_harvester_worker()
-    resource_descriptor = worker.get_resource(
-        harvestable_resource.unique_identifier,
-        harvestable_resource.remote_resource_type,
-        harvesting_session_id=harvesting_session_id)
-    worker.update_geonode_resource(resource_descriptor)
+    worker.update_geonode_resource(
+        harvestable_resource_id, harvesting_session_id)
```

Comment:
What is the reason for this change?

The issues description stated clearly that the focus should be on finishing the implementation of the `update_geonode_resource` using the already implemented structure.

I would prefer to have two methods with clear responsibilities over a single one.

The original approach, which was something like:

```python
# first extract the information from the remote service and create a resource descriptor
resource_descriptor = worker.get_resource()

# then use that resource descriptor to create/update the geonode resource
worker.update_geonode_resource(resource_descriptor)
```

seems cleaner and easier to test than this proposal:

```python

# everything done inside this method
worker.update_geonode_resource()
```

Unless you have some strong motive for changing the original implementation, I'd like this to be reverted

Reply:
Updated @ricardogsilva